Code/autopkglib/URLDownloader.py

```diff
@@ -254,6 +262,21 @@ def main(self):
                 if re.search(r'URL returned error: [0-9]+$', curlerr):
                     header['http_result_code'] = curlerr[curlerr.rfind(' ')+1:]
 
+        # Attempt to fail if there is a bad HTTP Status code.
+        # Sometimes, curl successfully exits, having not downloaded
+        # anything useful (e.g. we received a 404 message code, but 
+        # got some HTML, but this is flagged as success)
+        self.output( '%s has HTTP Status code %s' % ( self.env['url'], header['http_result_code'] ), verbose_level = 3 )
+        if self.env['HTTP_FAILURE_CODE_MIN']:
+            failure_min = self.env['HTTP_FAILURE_CODE_MIN']
+            http_status = header['http_result_code']
+            if type(failure_min) != IntType:
+                failure_min = atoi(failure_min)
+            if type(http_status) != IntType:
+                http_status = atoi(http_status)
```

Comment:
I just stumbled across this because I'm having numerous issues where my downloads appear to actually be small html source containing messages.

I'm curious about the use of `string.atoi` here. It was deprecated in Python 2.0 (October, 2000).  You should be able to replace these four lines with the following (no need to check the type, if they're already `int`, they'll just be passed through):

    failure_min = int(failure_min)
    http_status = int(http_status)